### PR TITLE
Chore: move from xorm to sqlx apikey store

### PR DESCRIPTION
**Pull request by ying-jeanne**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/services/sqlstore/sqlstore.go [hunk 1]
```diff
@@ -296,7 +296,7 @@ func (ss *SQLStore) buildConnectionString() (string, error) {
 		}
 
 		if ss.Cfg.IsFeatureToggleEnabled("mysqlAnsiQuotes") || ss.Cfg.IsFeatureToggleEnabled("newDBLibrary") {
-			cnnstr += "&sql_mode='ANSI_QUOTES'"
+			cnnstr += "&sql_mode='ANSI_QUOTES'&parseTime=true"
```
Comment (kylebrandt) [inline pkg/services/sqlstore/sqlstore.go:299]: Same as comment on prefs PR, mysqlAnsiQuotes should perhaps be independent

File: pkg/services/sqlstore/sqlutil/sqlutil.go [hunk 2]
```diff
@@ -28,7 +28,7 @@ func MySQLTestDB() TestDB {
 	if port == "" {
 		port = "3306"
 	}
-	conn_str := fmt.Sprintf("grafana:password@tcp(%s:%s)/grafana_tests?collation=utf8mb4_unicode_ci&sql_mode='ANSI_QUOTES'", host, port)
+	conn_str := fmt.Sprintf("grafana:password@tcp(%s:%s)/grafana_tests?collation=utf8mb4_unicode_ci&sql_mode='ANSI_QUOTES'&parseTime=true", host, port)
```
Comment (bergquist) [inline pkg/services/sqlstore/sqlutil/sqlutil.go:31]: Is this something we want to enable? If we do I would assume we do it for the none test connection as well as we would be testing the behavior with parseTime enabled.

Comment (ying-jeanne) [reply pkg/services/sqlstore/sqlutil/sqlutil.go:31]: yes, when the feature toggle is present, we would add also parsetime in the connection string for the non test connection https://github.com/grafana/grafana/pull/53434/files#diff-8d831fba1ec109fdc2ba8bcd44638e3d4def1b8fb725f874c2e043965ebb179aR302

File: pkg/services/apikey/apikeyimpl/sqlx_store.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+	if orgID != -1 {
+		err = ss.sess.Select(
+			ctx, &result, "SELECT * FROM api_key WHERE service_account_id IS NULL AND org_id = ? ORDER BY name ASC", orgID)
+	} else {
+		err = ss.sess.Select(
+			ctx, &result, "SELECT * FROM api_key WHERE service_account_id IS NULL ORDER BY name ASC")
+	}
+	if err != nil {
+		_ = err
+		// TODO: return error
```
[the remark this answers is not in the document]
Comment (ying-jeanne) [reply pkg/services/apikey/apikeyimpl/sqlx_store.go]: done

File: pkg/services/apikey/apikeyimpl/sqlx_store.go [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+		Key:              cmd.Key,
+		Created:          updated,
+		Updated:          updated,
+		Expires:          expires,
+		ServiceAccountId: nil,
+		IsRevoked:        &isRevoked,
+	}
+
+	t.Id, err = ss.sess.ExecWithReturningId(ctx,
+		`INSERT INTO api_key (org_id, name, role, "key", created, updated, expires, service_account_id, is_revoked) VALUES (?, ?, ?, ?, ?, ?, ?, ?,?)`, t.OrgId, t.Name, t.Role, t.Key, t.Created, t.Updated, t.Expires, t.ServiceAccountId, t.IsRevoked)
```
Comment (Jguer) [inline pkg/services/apikey/apikeyimpl/sqlx_store.go:106]:
```suggestion
		`INSERT INTO api_key (org_id, name, role, "key", created, updated, expires, service_account_id, is_revoked) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?)`, t.OrgId, t.Name, t.Role, t.Key, t.Created, t.Updated, t.Expires, t.ServiceAccountId, t.IsRevoked)
```

**Review by Jguer — APPROVED** [review]

**Pull request merged**